lib/services/location_search.dart

```diff
@@ -1,116 +1,85 @@
-import 'package:concordia_navigation/models/course.dart';
-import 'package:concordia_navigation/providers/calendar_data.dart';
+import 'package:concordia_navigation/models/indoor/indoor_location.dart';
+import 'package:concordia_navigation/models/outdoor/outdoor_location.dart';
+import 'package:concordia_navigation/models/uni_location.dart';
+import 'package:concordia_navigation/providers/indoor_data.dart';
+import 'package:concordia_navigation/screens/indoor_page.dart';
+import 'package:concordia_navigation/services/search.dart';
 import 'package:concordia_navigation/storage/app_constants.dart' as constants;
 import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
 import 'package:concordia_navigation/providers/map_data.dart';
-import 'dart:convert' show json;
-import 'package:flutter/services.dart' show rootBundle;
 
 /*This class extends Search Delegate class implemented by flutter.
 It will be called when the user clicks on the search button in the Appbar.
 */
 class LocationSearch extends SearchDelegate {
-  static List classrooms;
-  final recentRooms = [
-    'HALL BUILDING',
-    'H837',
-    'MB1.437'
-  ]; // for demonstration purposes
+  final bool isFirst;
 
-  static Future<List> loadJson() async =>
-      json.decode(await rootBundle.loadString('assets/destinations.json'));
+  LocationSearch(this.isFirst);
 
   ///This method returns suggested locations to the user, in this case Loyola and SGW campus.
   @override
   Widget buildSuggestions(BuildContext context) {
     final suggestionList = query.isEmpty
-        ? recentRooms
-        : classrooms.where((p) => p.contains(query.toUpperCase())).toList();
-
-    //TODO: make it clear that the first item is from the users calendar
-    CalendarData calendar = Provider.of<CalendarData>(context, listen: false);
-    List<Course> nextClasses = calendar.schedule?.nextClasses(days: 7);
-    if (nextClasses != null &&
-        nextClasses.isNotEmpty &&
-        nextClasses.first.filteredLocation() != "N/A") {
-      var next =
-          nextClasses.first.filteredLocation() + " [NEXT CLASS LOCATION]";
-      // Avoid duplicates on widget rebuild
-      if (!suggestionList.contains(next)) {
-        suggestionList.insert(0, next);
-      }
-    }
```

Comment:
#30 describes the need to ensure your next class is viewable and can be navigated to. The calendar schedule page is a whole week list. We should keep this entry in the location suggestion. Double check the epic's AT criterias. Plus its just a convenient and easy feature.

Reply:
We removed this feature since we did not see it was useful. The class calendar is is not that complicated to get to, we made it very easy to see next class locations.